File: backend/app/schemas/base_laboratorio.py
from pydantic import BaseModel


class BaseLaboratorioResponse(BaseModel):
    id: int
    periodo: str

    usuarios_estudiantes: int
    usuarios_profesores: int
    usuarios_externos: int

    servicios_solicitados: int
    servicios_atendidos: int

    promedio_horas_uso: float
    satisfaccion: float

    equipos_disponibles: int
    equipos_fuera_servicio: int

    incidentes: int

    class Config:
        from_attributes = True


class BaseLaboratorioStatsResponse(BaseModel):
    usuarios_totales: int
    servicios_solicitados: int
    servicios_atendidos: int
    equipos_disponibles: int
    incidentes: int
    tasa_atencion: float
    disponibilidad_equipos: float
    incidentes_por_100: float
    satisfaccion_promedio: float
    promedio_horas_uso: float


class BaseLaboratorioTrendDataPoint(BaseModel):
    periodo: str
    servicios_solicitados: int
    servicios_atendidos: int


class BaseLaboratorioUserDistribution(BaseModel):
    estudiantes: int
    profesores: int
    externos: int

File: frontend/src/models/BaseLaboratorio.ts
export interface BaseLaboratorio {
  id: number;
  periodo: string;
  usuarios_estudiantes: number;
  usuarios_profesores: number;
  usuarios_externos: number;
  servicios_solicitados: number;
  servicios_atendidos: number;
  promedio_horas_uso: number;
  satisfaccion: number;
  equipos_disponibles: number;
  equipos_fuera_servicio: number;
  incidentes: number;
}

export interface BaseLaboratorioStats {
  usuarios_totales: number;
  servicios_solicitados: number;
  servicios_atendidos: number;
  equipos_disponibles: number;
  incidentes: number;
  tasa_atencion: number;
  disponibilidad_equipos: number;
  incidentes_por_100: number;
  satisfaccion_promedio: number;
  promedio_horas_uso: number;
}

export interface BaseLaboratorioTrendDataPoint {
  periodo: string;
  servicios_solicitados: number;
  servicios_atendidos: number;
}

export interface BaseLaboratorioUserDistribution {
  estudiantes: number;
  profesores: number;
  externos: number;
}
